### PR TITLE
Error handling with anyhow, better error management

### PR DESCRIPTION
**Pull request by medzernik**
Error handling with the anyhow crate

### CODE REVIEW

File: src-tauri/src/main.rs [hunk 1]
```diff
@@ -242,7 +254,10 @@ async fn main() {
                 )
                 .max_file_size(Byte::from_unit(16_f64, ByteUnit::KiB).unwrap().into())
                 .targets([
-                    tauri_plugin_log::LogTarget::Folder(LogDirPath::file_name()),
+                    tauri_plugin_log::LogTarget::Folder(
+                        LogDirPath::file_name()
+                            .unwrap_or_else(|err| panic!("Error getting log folder name {}", err)),
```
[the remark this answers is not in the document]
Comment (medzernik) [reply src-tauri/src/main.rs:259]: I think if you do just expect, you lose the "err" value. This way, both are preserved (kind of like a nicer unwrap).

I may be wrong though

File: wooting-macro-backend/src/config.rs [hunk 2]
```diff
@@ -58,85 +63,121 @@ pub trait ConfigFile: Default + serde::Serialize + for<'de> serde::Deserialize<'
...
+            path = {
+                let conf_dir: Result<PathBuf> = match dirs::config_dir() {
+                    Some(config_path) => Ok(config_path),
+                    None => Err(anyhow::Error::msg(
+                        "Cannot find config directory, cannot proceed.",
+                    )),
+                };
+                conf_dir?.join(CONFIG_DIR)
+            };
+        }
```
Comment (simon-wh) [inline wooting-macro-backend/src/config.rs:123]: What's the real world situation in where the config_dir will fail to retrieve? This seems more like an unwrap and panic situation than bubbling up an error

Comment (simon-wh) [reply wooting-macro-backend/src/config.rs:123]: What happens when it panics? Cus in some cases you should if it's a hard error and be able to determine the panic state

File: src-tauri/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (medzernik) [reply src-tauri/src/main.rs:259]: image

should be the same, will change all the panics like that to expect. Dunno why I thought the err value gets lost.

File: wooting-macro-backend/src/config.rs (hunk 2 above)
Comment (medzernik) [reply wooting-macro-backend/src/config.rs:123]: > What happens when it panics? Cus in some cases you should if it's a hard error and be able to determine the panic state

It should bubble up and then it's decided by the upper functions what to do. I think i made it panic in the end in ``src-tauri``, where it's used most.

File: src-tauri/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (medzernik) [reply src-tauri/src/main.rs]: Replaced [8c4459e](https://github.com/WootingKb/wooting-macros/pull/179/commits/8c4459e5667b396a4f48b13830ebc4d883bec607)

File: wooting-macro-backend/src/config.rs (hunk 2 above)
Comment (medzernik) [reply wooting-macro-backend/src/config.rs:122]: image

image

Yeah, these errors get .expected later on, it's just there so the caller can figure out whether to panic outright. Not sure if the best approach but figured it's a better one than just randomly panicking somewhere deeper in library. 

We don't really handle cases where there's no XDG directories setup, we just panic.

**Pull request merged**